### PR TITLE
Shared mode

### PR DESCRIPTION
**Pull request by jluhrsen**

### CODE REVIEW

**Comment by jluhrsen** [comment]
using this to test that shared mode is deployed like we expect if a manifest exists with the proper configMap

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 1]
```diff
@@ -795,13 +795,13 @@ spec:
...
-          # Check to see if ovs is provided by the node. This is only for upgrade from 4.5->4.6 or
-          # openshift-sdn to ovn-kube conversion
-          if grep -q OVNKubernetes /etc/systemd/system/ovs-configuration.service ; then
+          if [ "{{.OVN_GATEWAY_MODE}}" == "shared" ]; then
             gateway_mode_flags="--gateway-mode shared --gateway-interface br-ex"
+          elif [ "{{.OVN_GATEWAY_MODE}}" == "local" ]; then
+            gateway_mode_flags="--gateway-mode local --gateway-interface br-ex"
           else
-            gateway_mode_flags="--gateway-mode local --gateway-interface none"
+            echo "Invalid OVN_GATEWAY_MODE: \"{{.OVN_GATEWAY_MODE}}\". Must be \"local\" or \"shared\"."
```
Comment (aojea) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml:803]: how can we have a wrong value here?
shouldn't this be validated in the operator?

File: pkg/network/ovn_kubernetes.go [hunk 2]
```diff
@@ -212,6 +215,26 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	return objs, nil
 }
 
+// returns the value of mode found in the openshift-ovn-kubernetes/gateway-mode-config configMap
+// if it exists, otherwise returns whatever the global OVN_GATEWAY_MODE is set to (shared)
+func GetGatewayMode(kubeClient client.Client) string {
```
Comment (aojea) [inline pkg/network/ovn_kubernetes.go:220]: is this not a CRD option already?

**Pull request closed without merging**